shared/simplestreams/products.go

```diff
@@ -130,6 +130,10 @@ func (s *Products) ToLXD() ([]api.Image, map[string][][]string) {
 					return fmt.Errorf("No LXD image fingerprint found")
 				}
 
+				if downloads[fingerprint] != nil {
```

Comment:
whats this for?

Reply:
It is a bit unclear but the images link to the version item with `FileType==incus.tar.xz` or `FileType==lxd.tar.xz` seem to have the same fingerprint so we end up having an error later in `getImages()` ... So I used the `downloads` map to check that the added fingerprints are unique..